SONARJAVA-5294 Create rule S7185: @eventlistener methods should have one parameter (#4618)
SonarSource/rspec: rules/S7185/java/rule.adoc

== Why is this an issue?

Spring provides the `@EventListener` annotation to make a method handle some specific types of events.
This allows to skip the implementation of the `ApplicationListener` interface, making it easier to handle events.

> However, the `@EventListener` annotation can only be used on methods that have at most one parameter, which should be the specific event that we want to handle.

Reviewer: Let's add a line break and remove the however.

```suggestion
The `@EventListener` annotation can only be used on methods that have at most one parameter, which should be the specific event that we want to handle.
```

> For multiple types of events, we can use the `classes` argument of the annotation, and provide a list of classes of events that we want to handle.

Reviewer: ```suggestion
To listen to several types of events, use the `classes` argument of the `@EventListener` annotation.
```

== Why is this an issue?

> Spring provides the `@EventListener` annotation to make a method handle some specific types of events.

Reviewer: ```suggestion
Spring provides the `@EventListener` annotation to simplify event handling by allowing methods to handle events.
```
